src/transformers/modeling_bart.py

```diff
@@ -995,6 +994,12 @@ def _reorder_cache(past, beam_idx):
         past = ((new_enc_out, new_enc_mask), reordered_past)
         return past
 
+    def encode(self, encoder_input_ids, **kwargs):
+        # get only encoder output ids
+        encoder_output_ids = self.model.encoder(encoder_input_ids, **kwargs)
```

Comment:
these aren't usually `ids`. Suggest rename `encoder_outputs`
Love the refactor though!